src/communication/context.cpp

```diff
@@ -78,7 +78,10 @@ bool ClientContext::use_ssl() { return use_ssl_; }
 
 ServerContext::ServerContext(const std::string &key_file, const std::string &cert_file, const std::string &ca_file,
                              bool verify_peer) {
-  ctx_.emplace(boost::asio::ssl::context::tls_server);
+  ctx_.emplace(boost::asio::ssl::context::sslv23_server);
+  // NOLINTNEXTLINE(hicpp-signed-bitwise)
+  ctx_->set_options(boost::asio::ssl::context::default_workarounds | boost::asio::ssl::context::no_sslv2 |
```

Comment:
Don't we need TLS instead of SSL?

Reply:
I am not entirely sure what doe flags imply, so I will check it out.

Reply:
I've left our boost::asio::context suffix, but here are the flags and what I think which ones we should apply:
- `sslv23_server` will be converted to `tls_server` since we want to use a modern version of SSL a TLS
- `default_workarounds` sets a group of default options for a specific OpenSSL version, I would keep this since the options were defined by the maintainer themselves
- `no_sslv2`  translates to `SSL_OP_NO_SSLv2` which disables ssl v2 and that's recommended, I suggest disabling ssl v3 as well since that's no longer secure
- `single_dh_use` translates to `SSL_OP_SINGLE_DH_USE` and that forces the older OpenSSL version (prior 1.0.2) to use  a new key for each session, we will leave for centos 7